sql/migrate/migrate.go

```diff
@@ -159,6 +159,23 @@ type (
 		Name() string
 	}
 
+	// DirUnraveler wraps the UnravelDir method.
+	DirUnraveler interface {
+		// UnravelDir returns a set of files from the given Dir to be executed on a database.
+		UnravelDir(Dir) []File
+	}
+
+	// FileUnraveler wraps the UnravelFile method.
+	FileUnraveler interface {
+		UnravelFile(File) []string
+	}
```

Comment:
Do you mind explaining why these 2 interfaces are needed? Is it here because you have a WIP code or does it help organizing your thoughts for now? 

In general, it's hard (for me) to see the benefit of interfaces without seeing how they are used (consumed or implemented). 

Last comment about this - Why did you choose the term/word Unraveler? it feels less common and not intuitive for usage (and verbal discussions), but again, it's hard for me to judge without seeing implementations or usage of it.